sqlparse/sql.py

```diff
@@ -257,37 +257,25 @@ def token_prev(self, idx, skip_ws=True, skip_cm=False):
         """Returns the previous token relative to *idx*.
 
         If *skip_ws* is ``True`` (the default) whitespace tokens are ignored.
+        If *skip_cm* is ``True`` comments are ignored.
         ``None`` is returned if there's no previous token.
         """
-        if idx is None:
-            return None, None
-        idx += 1  # alot of code usage current pre-compensates for this
-        funcs = lambda tk: not ((skip_ws and tk.is_whitespace()) or
-                                (skip_cm and imt(tk, t=T.Comment, i=Comment)))
-        return self._token_matching(funcs, idx, reverse=True)
+        return self.token_next(idx, skip_ws, skip_cm, _reverse=True)
 
-    # TODO: May need to implement skip_cm for upstream changes.
     # TODO: May need to re-add default value to idx
-    def token_next(self, idx, skip_ws=True, skip_cm=False):
+    def token_next(self, idx, skip_ws=True, skip_cm=False, _reverse=False):
         """Returns the next token relative to *idx*.
 
         If *skip_ws* is ``True`` (the default) whitespace tokens are ignored.
+        If *skip_cm* is ``True`` comments are ignored.
         ``None`` is returned if there's no next token.
         """
         if idx is None:
             return None, None
         idx += 1  # alot of code usage current pre-compensates for this
-        try:
-            if not skip_ws:
-                return idx, self.tokens[idx]
-            else:
-                while True:
-                    token = self.tokens[idx]
-                    if not token.is_whitespace():
-                        return idx, token
-                    idx += 1
-        except IndexError:
-            return None, None
+        funcs = lambda tk: not ((skip_ws and tk.is_whitespace()) or
+                                (skip_cm and imt(tk, t=T.Comment, i=Comment)))
```

Comment:
I had the same change in an [upstream branch](https://github.com/vmuriart/sqlparse/blob/b4424963fd1ed631daf2cb9adc70906b30f56494/sqlparse/sql.py#L278) :+1: 
The `pr` that had the original change was before `skip_cm` was implemented, I left the comment to remind myself to implement it later since other changes depend on them.